modules/kmm-signing-a-prebuilt-driver-container.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-signing-a-prebuilt-driver-container_{context}"]
+= Signing a pre-built driver container
+
+Use this procedure if you have a pre-built image, such as an image distributed by a hardware vendor
+or is already built elsewhere.
+
+The YAML file shown below adds the public/private key-pair as secrets with the required key names - `key` for the private key, `cert` for the public key. It will then pull down the `unsignedImage` image, open it, sign the kernel modules listed in `filesToSign`, add them back and push the resulting image as `containerImage`.
+
+Kernel Module Management (KMM) should then deploy the DaemonSet that loads the signed kmods onto all the nodes
+that match the selector. The driver containers should run successfully on any nodes that have the public key in their MOK database,
+and any nodes that are not secure-boot enabled, which will ignore the signature. They should fail to load on any that have secure-boot enabled but do not have that key in their MOK database.
+
+.Prerequisites
+
+* The `keySecret` and `certSecret` secrets have been created.
+
+.Procedure
+
+. Apply the YAML file:
++
+[source,yaml]
+----
+---
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: Module
+metadata:
+  name: example-module
+spec:
+  moduleLoader:
+    serviceAccountName: default
+    container:
+      modprobe:
+        # the name of the kmod to load
```

Comment:
I think callouts would be more consistent with the rest of docs than YAML comments.